src/gaxios.ts

```diff
@@ -219,13 +270,14 @@ export class Gaxios {
     }
     opts.method = opts.method || 'GET';
 
-    const proxy = loadProxy();
+    const proxy = getProxy(opts.url);
     if (proxy) {
+      loadProxy();
```

Comment:
what if you make the first line of `getProxy` call `loadProxy`, just to ensure that it's been called; this will probably address the issue in tests and feels less weird.